multiversx_sdk_core/transaction.py

```diff
@@ -1,156 +1,131 @@
-
-        self.version: ITransactionVersion = version or TRANSACTION_VERSION_DEFAULT
-        self.options: ITransactionOptions = options or TRANSACTION_OPTIONS_DEFAULT
-
-        self.signature: ISignature = bytes()
-        self.guardian_signature: ISignature = bytes()
-
-    def serialize_for_signing(self) -> bytes:
-        dictionary = self.to_dictionary(with_signature=False)
+    def __init__(self,
+                 sender: str,
+                 receiver: str,
+                 gas_limit: int,
+                 chain_id: str,
+                 nonce: Optional[int] = None,
+                 amount: Optional[int] = None,
+                 sender_username: Optional[str] = None,
+                 receiver_username: Optional[str] = None,
+                 gas_price: Optional[int] = None,
+                 data: Optional[bytes] = None,
+                 version: Optional[int] = None,
+                 options: Optional[int] = None,
+                 guardian: Optional[str] = None,
+                 signature: Optional[bytes] = None,
+                 guardian_signature: Optional[bytes] = None
+                 ) -> None:
+        self.chainID = chain_id
+        self.sender = sender
+        self.receiver = receiver
+        self.gas_limit = gas_limit
+
+        self.nonce = nonce or 0
+        self.amount = amount or 0
+        self.data = data or bytes()
+        self.signature = signature or bytes()
+
+        self.sender_username = sender_username or ""
+        self.receiver_username = receiver_username or ""
+
+        self.gas_price = gas_price or TRANSACTION_MIN_GAS_PRICE
+        self.version = version or TRANSACTION_VERSION_DEFAULT
+        self.options = options or TRANSACTION_OPTIONS_DEFAULT
+
+        self.guardian = guardian or ""
+        self.guardian_signature = guardian_signature or bytes()
+
+
+class TransactionComputer:
+    def __init__(self, address_converter: IAddressConverter) -> None:
+        self.address_converter = address_converter
+
+    def compute_transaction_fee(self, transaction: Transaction, network_config: INetworkConfig) -> int:
+        move_balance_gas = network_config.min_gas_limit + len(transaction.data) * network_config.gas_per_data_byte
+        if move_balance_gas > transaction.gas_limit:
+            raise NotEnoughGasError(transaction.gas_limit)
+
+        fee_for_move = move_balance_gas * transaction.gas_price
+        if move_balance_gas == transaction.gas_limit:
+            return int(fee_for_move)
+
+        diff = transaction.gas_limit - move_balance_gas
+        modified_gas_price = transaction.gas_price * network_config.gas_price_modifier
+        processing_fee = diff * modified_gas_price
+
+        return int(fee_for_move + processing_fee)
+
+    def compute_bytes_for_signing(self, transaction: Transaction) -> bytes:
+        dictionary = self._to_dictionary(transaction)
         serialized = self._dict_to_json(dictionary)
```

Comment:
these 2 functions looks like a `JSONSerializer` for transaction.
I'm thinking if we can have the protobuf Transaction struct serialized as json directly from their library or something.

Maybe to evaluate for a future PR.

Reply:
Will look into this for a future PR.